### DISCUSSION TITLE
Wanting to integrate jwttoken auth with GraphQL

### DISCUSSION THREAD

**Discussion by pkellner** [vercel/next.js#13608] [Help]
The example https://github.com/lfades/next-with-apollo looks awesome for integrating GraphQL with next but I'm struggling with a best practices way of incorporating JWTTokens into that.  Years ago, there was a lot of discussion around login support but that seems to have been de-prioritized as examples.

Any URL's that have a good clean solution would be appreciated. I like the apollo solutions since there is a lot of following behind them.

**Comment by lfades** [comment]
First, check out one of our Authentication examples, we have auth examples for Auth0, magic.link, Userbase, Firebase, Passport.js and more.

Once you have decided for one of those, your app (and Apollo) will access to a login token that can be send to your GraphQL server as a cookie, or as the `Authorization` header, once that's done you can start making authenticated requests.

**Reply by Dannymx** [reply]
@lfades Is there any example where Next.js requests a JWT that is saved in a cookie then passed to Apollo Client? I've been trying to get a working example of `with-iron-session` and `with-apollo` but I'm stuck trying to pass my saved iron session to the Apollo Client as I can't run any hooks in `apolloClient.js`

**Reply by lfades** [reply]
@Dannymx I'm not sure what to do there tbh. We usually send the cookies with the request, so you won't need any hooks.

**Reply by pkellner** [reply]
@Dannymx Did you figure out how to do that? I'm back to working on that again.
